app/src/main/java/org/dslul/openboard/inputmethod/latin/settings/Settings.java

```diff
@@ -502,6 +503,7 @@ public static void setSecondaryLocales(final SharedPreferences prefs, final Stri
         prefs.edit().putString(PREF_SECONDARY_LOCALES_PREFIX + mainLocaleString.toLowerCase(Locale.ROOT), sb.toString()).apply();
     }
 
+    @RequiresApi(api = Build.VERSION_CODES.S)
```

Comment:
Actually all Android versions need to call this function.
Why did you add this? Is there some warning from Android Studio, or compilation error?

Reply:
There was indeed a warning in Android studio, but it didn't prevent the application from being compiled.
I discovered this by accident.

Reply:
Ah, I see that `KeyboardTheme.getThemeColors` also has `@RequiresApi`. It should better return `DynamicColors` only if api >= Build.VERSION_CODES.S, so
```diff
             case THEME_DYNAMIC:
-                return new DynamicColors(context, prefs);
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S)
+                    return new DynamicColors(context, prefs);
```
No else branch should be needed here.